### PR TITLE
Paper item demo

### LINKED ISSUE #383: Add example for paper-item in item templates
`<paper-item>` is a ready-made item wrapper with material design look. It covers plenty of simple cases.

Since `<vaadin-combo-box>` is also material-like by default, pointing to`<paper-item>` in the examples could be helpful for the users.

### PR DESCRIPTION
**Pull request by tomivirkki**
Fixes #383

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-combo-box/395)

### CODE REVIEW

**Comment by platosha** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

*[demo/common.html, line 190 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-KcXGqYWqd3bD3X3gy4A:-KcXGqYXvC3D0YhzNRE1:bqs645o) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/bd5b21dccec70c4f9e00e3d98f78bc0c526864ce/demo/common.html#L190)):*
> ```HTML
>     });
> 
>     var users = ({
> ```

The following data is quite heavy.

I would include only necessary data, for example: https://randomuser.me/api/?inc=name,email,picture&results=100

Also, it gives an array of users right away, no need to `map` through.

Also, maybe have a comment with this origin JSON API link, to credit them and to be able to quickly generate more data if needed in future?

---

*[demo/item-template.html, line 136 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-KcXHvP0RCB7Q3NghbW7:-KcXHvP1qN9vQG1l9XEm:b-xqlwcv) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/bd5b21dccec70c4f9e00e3d98f78bc0c526864ce/demo/item-template.html#L136)):*
> ```HTML
>     <demo-snippet>
>       <template>
>         <dom-module id="my-paper-item">
> ```

Maybe shorten this to use paper-item directly (and some inline styles), and avoid wrapping in another custom element?

We have a custom element example already.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395)*

**Comment by tomivirkki** [comment]
Review status: 2 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[demo/common.html, line 190 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-KcXGqYWqd3bD3X3gy4A:-KcXU8ClF6RL43Ix35At:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/bd5b21dccec70c4f9e00e3d98f78bc0c526864ce/demo/common.html#L190)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

The following data is quite heavy.

I would include only necessary data, for example: https://randomuser.me/api/?inc=name,email,picture&results=100

Also, it gives an array of users right away, no need to `map` through.

Also, maybe have a comment with this origin JSON API link, to credit them and to be able to quickly generate more data if needed in future?
</blockquote></details>

Done.

---

*[demo/item-template.html, line 136 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-KcXHvP0RCB7Q3NghbW7:-KcXUAglVNWv69cZqcht:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/bd5b21dccec70c4f9e00e3d98f78bc0c526864ce/demo/item-template.html#L136)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Maybe shorten this to use paper-item directly (and some inline styles), and avoid wrapping in another custom element?

We have a custom element example already.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395)*

**Comment by platosha** [comment]
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[demo/item-template.html, line 140 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-KcXUZywa-KAndcGBd2G:-KcXUZyxVDhnK8kGcEBr:b-ww5cms) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/1926601e7799eeaaca6456467ef8c4530d5c5e4b/demo/item-template.html#L140)):*
> ```HTML
>             <paper-item style="padding: 0">
>               
>               <div style="padding: 0 8px;">
> ```

It was `<paper-item-body>` before. Why change to `<div>`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395)*

**Comment by tomivirkki** [comment]
*[demo/item-template.html, line 140 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-KcXUZywa-KAndcGBd2G:-KcXdpZXBw4aYb0OczVN:b-7q8qmh) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/1926601e7799eeaaca6456467ef8c4530d5c5e4b/demo/item-template.html#L140)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

It was `<paper-item-body>` before. Why change to `<div>`?
</blockquote></details>

Done. Also switched back to paper-icon-item, needs the least extra styles.

The item wrappers in combo-box have 13px 16px paddings. This causes issues with paper-item-body since it has 72px min-height (two-line version). In Jouni's demo this was worked around by using negative margins. Was going back and forth with this but ended up with overriding min-height from paper-item-body (to avoid magic numbers in the demo/docs).

Maybe the best option would be to not have paddings for the combo box item wrappers but instead have the 13px 16px paddings in the default template only?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395)*

**Comment by platosha** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[demo/item-template.html, line 140 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-KcXUZywa-KAndcGBd2G:-KcXkIx3rxjqBXLmlQpv:bpy4g81) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/1926601e7799eeaaca6456467ef8c4530d5c5e4b/demo/item-template.html#L140)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Done. Also switched back to paper-icon-item, needs the least extra styles.

The item wrappers in combo-box have 13px 16px paddings. This causes issues with paper-item-body since it has 72px min-height (two-line version). In Jouni's demo this was worked around by using negative margins. Was going back and forth with this but ended up with overriding min-height from paper-item-body (to avoid magic numbers in the demo/docs).

Maybe the best option would be to not have paddings for the combo box item wrappers but instead have the 13px 16px paddings in the default template only?
</blockquote></details>

LGTM

IMO, default paddings work really well with simple inline HTML item templates, I would keep them. We should add a CSS mixin for the item to allow removing the paddings for such cases.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-:-KcXl0SsLpazMC06PE5t:bnfp4nl)*

**Comment by platosha** [comment]
Please fix the linter errors.

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-:-KcXlAebOKxCsI2J8L6j:b8ztqam)*

**Comment by tomivirkki** [comment]
Oops, fixed linter errors :)

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-:-KcXx1X05XrvriozlwJB:b-hc02dg)*

**Comment by limonte** [comment]
:lgtm:

---

Review status: 4 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-:-KcXxvGOZRkB1tjJ6A2m:bnfp4nl)*

**Comment by platosha** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/395#-:-KcY2CNhndjlKLRWtZ0d:bnfp4nl)*
